### ISSUE TITLE
QueryBuilder (or DatabaseCompartment) should be accessible

### ISSUE THREAD

**Issue by simplysoft** [erickok/RxCupboard#6]
In order to perform some advanced queries, the query builder should be accessible from RxDatabase

**Comment by erickok** [comment]
I see you mean use the DatabseCompartment's query builder feature and then still get the results as Rx stream? Let me see...

**Comment by simplysoft** [comment]
yes, currently the RxDatabase only exposes the simple query methods, e.g. limiting of the result is not possible. The query builder is also needed e.g. to use an agregation function, e.g. select max(*) from table

**Comment by erickok** [comment]
Okay, so I actually forgot :blush: but you can already pass a prepared query to RxDatabase. For example:

``` java
        DatabaseCompartment.QueryBuilder<TestEntity> query =
                cupboard.withDatabase(db).query(TestEntity.class).withSelection("_id <= ?", Integer.toString(5));
        rxDatabase.query(query).subscribe(entity -> doSomeThingWith(entity));
```

See https://github.com/erickok/RxCupboard/blob/master/library/src/androidTest/java/nl/nl2312/rxcupboard/QueryTest.java#L86 for the method that tests this.

If you have any suggestion for changes or improvements, let me know.

**Comment by simplysoft** [comment]
Thanks. I'd rather say that this is a workaround then a nice way to do it. The problem is that this approach uses its own independent database compartment and not necessary the one rxdatabase is using, which could potentially result in some surprising results. 
I think it would be much cleaner to just expose the method `public <T> QueryBuilder<T> query(Class<T> entityClass)`from the DatabaseCompartment in RxDatabase

**Comment by erickok** [comment]
Hmm I partly agree. The method name query() is already reserved though for applying a query. How about a `public <T> QueryBuilder<T> query(Class<T> entityClass)`, which you would use like:

``` java
rxDatabase.query(
    rxDatabase.builder(TestEntity.class).withSelection("_id <= ?", Integer.toString(5)))
        .subscribe(entity -> doSomeThingWith(entity));
```

Alternative naming options: `queryBuilder` (what it returns) or `buildQuery` (verb like the others).

**Comment by erickok** [comment]
Released with 0.6 version (uploading to Maven Central now).

**Comment by simplysoft** [comment]
:+1: 
thank for the enhancement & publishing of a new version. (like your final decision about the method name.)
